contracts/tg4-engagement/src/contract.rs

```diff
@@ -1616,18 +1629,51 @@ mod tests {
         do_instantiate(deps.as_mut());
         let mut env = mock_env();
 
+        // register a hook, to check for half life side effects
+        let contract1 = String::from("hook1");
+        let contract2 = String::from("hook2");
+
+        let admin_info = mock_info(INIT_ADMIN, &[]);
+        let hook_msg = ExecuteMsg::AddHook {
+            addr: contract1.clone(),
+        };
+        let hook_msg2 = ExecuteMsg::AddHook {
+            addr: contract2.clone(),
+        };
+        let _ = execute(deps.as_mut(), mock_env(), admin_info.clone(), hook_msg).unwrap();
+        let _ = execute(deps.as_mut(), mock_env(), admin_info, hook_msg2).unwrap();
+
         // end block just before half life time is met - do nothing
         env.block.time = env.block.time.plus_seconds(HALFLIFE - 2);
         assert_eq!(end_block(deps.as_mut(), env.clone()), Ok(Response::new()));
         assert_users(&deps, Some(USER1_POINTS), Some(USER2_POINTS), None, None);
 
         // end block at half life
         env.block.time = env.block.time.plus_seconds(HALFLIFE);
-        let expected_reduction = points_reduction(USER1_POINTS) + points_reduction(USER2_POINTS);
+        let expected_reduction_user1 = points_reduction(USER1_POINTS);
+        let expected_reduction_user2 = points_reduction(USER2_POINTS);
+        let expected_reduction = expected_reduction_user1 + expected_reduction_user2;
         let evt = Event::new("halflife")
             .add_attribute("height", env.block.height.to_string())
             .add_attribute("reduction", expected_reduction.to_string());
-        let resp = Response::new().add_event(evt);
+        let msg = MemberChangedHookMsg {
```

Comment:
Nice. Looks good.

It would be good to cover the same case with migrate.
Doesn't matter if there was/wasn't an untriggered half-life before. (no need to simulate that)
Point is migrate should return the selected set of MembeDiff messages.

Reply:
Yes. Will do.